modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -313,6 +315,7 @@ protected void initChannel(Channel ch) throws Exception {
                 ch.pipeline().addLast("encoder_compress", new HttpContentCompressor(handlingSettings.getCompressionLevel()));
             }
             ch.pipeline().addLast("request_creator", requestCreator);
+            ch.pipeline().addLast("response_creator", responseCreator);
```

Comment:
Should we only add this when `handlingSettings.isCompression() == true`?

Reply:
I don't think so. It does not seem necessary to create a separate codepath for this pretty niche scenario.

Reply:
Don't we by default have compression off with tls enabled?

Reply:
You're right. I had forgotten that we disable compression with TLS unless it is manually configured. I still think the single codepaths is the correct approach unless for some reason a nightly benchmark indicates we need a change.

Reply:
I don't think this will show in benchmarks it's a small thing relative to the general noise in them for sure, but to me it seems strange to add general overhead for a workaround that's very specific to a concrete issue with the Netty compression implementation? (not too important to me, but still seems like it would also make it clearer why we're doing this).

Reply:
Because it is nice to have a consistent manner in which data flows down the pipeline instead of allowing the various features to create different combinations that must be tracked when investigating issues.

And this PR does not go all the way, but splitting data up when sending is on the list of TODOs anyway. The fact that we batch compress and batch TLS encode on the HTTP outbound pipeline is not ideal. And a reason why the nio transport has outperformed the netty transport on scroll benchmarks. So I imagine there will be some follow-up work at some point to incrementally send data down the pipeline and flush, waiting for writability before sending more down.